### DISCUSSION TITLE
Does ipopt do variable eliminates?

### DISCUSSION THREAD

**Discussion by broesdecat** [coin-or/Ipopt#436] [Q&A] [answered]
Hey,

I have a model with some with some variables on which there is an equality constraint (e.g. x1==x2), and only one of them has been given a starting value (say x1). I notice that performance of my model is a lot better if I assign the same starting value to both x1 and x2.
So that makes me wonder: is there something in the solver that eliminates variables (substituting x2 for x1), and does it take starting value into account? By default or maybe behind an option?

Thanks in advance,
Broes

**Comment by svigerske** [comment, marked as answer]
There is essentially no presolve happening. The only thing that happens is that Ipopt hides fixes variables from the internal formulation by default.

So given the same starting value for variables that you know will have to have the same value in a feasible solution should be good.
